### PR TITLE
Improve serialization of floating point values

### PR DESCRIPTION
**Pull request by GrishaKirilin**
Hi. Thank you for such a convenient library! I started to use it in almost all my cmdl utilities. 
In particular, I use it to parse configuration files that may contain floating-point values.

Here is my small improvement of the serialization of floating point numbers.

### CODE REVIEW

File: argh.h [hunk 1]
```diff
@@ -382,8 +388,11 @@ namespace argh
          auto optIt = params_.find(trim_leading_dashes(name));
          if (params_.end() != optIt)
             return string_stream(optIt->second);
-      }      
+      }
       std::ostringstream ostr;
+      using value_type = typename std::decay<T>::type;
```
Comment (adishavit) [inline argh.h]: Please add a comment explaining what you are doing in the next 3 lines and why.

File: argh.h [hunk 2]
```diff
@@ -367,6 +370,9 @@ namespace argh
          return string_stream(optIt->second);
 
       std::ostringstream ostr;
+      using value_type = typename std::decay<T>::type;
+      if (std::is_floating_point<value_type>::value)
+          ostr << std::setprecision(std::numeric_limits<value_type>::max_digits10);
```
[the remark this answers is not in the document]
Comment (GrishaKirilin) [reply argh.h]: It is certainly so. It allows to get rid of iomanip header.

File: argh.h (hunk 1 above)
Comment (GrishaKirilin) [reply argh.h]: Should I add a comment in the code? References should be removed from the type `T` in order to use type trait `is_floating_point`. Unfortunately, numeric limits (sometimes) cannot be correctly determined for arrays, thus one should decay the type before.

But now I am not sure that it is actually necessary, the simplest way is to set the maximal precision like `ostr.precision(std::numeric_limits<long double>::max_digits10);` It is only one line and it solves all the problems. What do you think?

**Comment by adishavit** [comment]
I think that should work regardless of the floatiness. Just add the one line.

**Comment by GrishaKirilin** [comment]
Done. One line per case, no type traits.

**Comment by adishavit** [comment]
Much appreciated!

**Pull request merged**